### DISCUSSION TITLE
[Feature] fully support by-ref types in curried functions

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1235] [General]
Due to the nature of functional programming, it's possible to partially apply function with by ref types as long they don't leave the stack (E.g the curried function being returned). Function is simply a processing unit and these detail of how to process things are described in our code, unlike imperative programming where state mutates, it's possible to contain things well in units of process therefore having things that seems like they would be leaving the stack like curried functions but will stay in stack (they often do anyways through inlining)

**Comment by vzarytovskii** [comment]
Please, create a suggestion (or find an existing one), the process is described in a README for this repo

**Reply by cartermp** [reply]
Yep, to add to this, there's a suggestion here already.
